app/src/androidTest/java/com/android/voyageur/ui/notifications/AndroidNotificationProvidertest.kt

```diff
@@ -0,0 +1,189 @@
+package com.android.voyageur.ui.notifications
+
+import android.app.NotificationManager
+import android.content.Context
+import android.content.Intent
+import android.os.SystemClock
+import androidx.core.app.NotificationCompat
+import androidx.test.core.app.ActivityScenario
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.rule.GrantPermissionRule
+import androidx.test.uiautomator.By
+import androidx.test.uiautomator.UiDevice
+import androidx.test.uiautomator.Until
+import com.android.voyageur.MainActivity
+import com.android.voyageur.R
+import junit.framework.Assert.assertNotNull
+import junit.framework.Assert.assertTrue
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class AndroidNotificationProviderTest {
+
+  @get:Rule
+  val permissionRule: GrantPermissionRule =
+      GrantPermissionRule.grant(android.Manifest.permission.POST_NOTIFICATIONS)
+
+  private lateinit var context: Context
+  private lateinit var notificationProvider: AndroidNotificationProvider
+  private lateinit var notificationManager: NotificationManager
+  private lateinit var uiDevice: UiDevice
+  private lateinit var activityScenario: ActivityScenario<MainActivity>
+
+  private val NOTIFICATION_TIMEOUT = 5000L
+  private val TEST_SENDER_NAME = "John Doe"
+  private val TEST_ACCEPTOR_NAME = "Jane Smith"
+
+  @Before
+  fun setup() {
+    context = ApplicationProvider.getApplicationContext()
+    notificationProvider = AndroidNotificationProvider(context)
+    notificationManager =
+        context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+    uiDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+
+    // Clear existing notifications
+    notificationManager.cancelAll()
+
+    // Initialize notification channel
+    NotificationHelper.createNotificationChannel(context)
+
+    // Launch MainActivity
+    val intent =
+        Intent(context, MainActivity::class.java).apply {
+          addFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK)
+        }
+    activityScenario = ActivityScenario.launch(intent)
+
+    // Ensure we're starting from the home screen
+    uiDevice.pressHome()
+  }
+
+  @After
+  fun tearDown() {
+    notificationManager.cancelAll()
+    activityScenario.close()
+  }
+
+  @Test
+  fun testNewFriendRequestNotification() {
+    // Wait for activity to be fully created
+    SystemClock.sleep(1000)
```

Comment:
Since we're removing all hardcoded strings, I suggest to use the same approach with "magic numbers" to maintain consistency throughout the code

Reply:
They are not really the same thing since we remove hardcoded strings such that we can have support for multiple languages, but the time intervals in the code are needed in any situation and so I believe they can be hardcoded since they do not impact any UI elements. Also the code reference you put is from tests and we can implement them anyway we want.